web/adventures/views.py

```diff
@@ -0,0 +1,21 @@
+from django.http import JsonResponse, HttpResponse
+from django.shortcuts import get_object_or_404
+from django.forms.models import model_to_dict
+from . import models
+
+
+def adventure_by_id(request, adventure_id):
+    # This endpoint only handles information retrieval, so only GET
+    # requests are allowed. Respond with 'METHOD NOT ALLOWED'.
+    if request.method != 'GET':
+        return HttpResponse(status=405)
+
+    # Adventure IDs must be integers, respond with BAD REQUEST if an
+    # invalid ID is given.
+    is_valid_id = all(char.isdigit() for char in adventure_id)
```

Comment:
My thinking was that it gives more specific information about the error to the caller than a 404 that could mean multiple things. Would that be reason enough? I agree it's nicer to handle with the regex, I was originally going to do it that way.